akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/Behaviors.scala

```diff
@@ -169,25 +171,35 @@ object Behaviors {
    * This type of Behavior wraps another Behavior while allowing you to perform
    * some action upon each received message or signal. It is most commonly used
    * for logging or tracing what a certain Actor does.
+   *
+   * @param id Used to identify when a tap is added again (using instance equality),
+   *           and the outermost/previous one should be removed. If not called recursively,
+   *           just pass a new `InterceptId` instance.
    */
   def tap[T](
     clazz:     Class[T],
     onMessage: Procedure2[ActorContext[T], T],
     onSignal:  Procedure2[ActorContext[T], Signal],
+    id:        WrappedBehaviorId,
     behavior:  Behavior[T]): Behavior[T] = {
     akka.actor.typed.scaladsl.Behaviors.tap[T](behavior)(
       (ctx, msg) ⇒ onMessage.apply(ctx.asJava, msg),
-      (ctx, sig) ⇒ onSignal.apply(ctx.asJava, sig))(ClassTag[T](clazz))
+      (ctx, sig) ⇒ onSignal.apply(ctx.asJava, sig),
+      id)(ClassTag[T](clazz))
   }
 
   /**
    * Behavior decorator that copies all received message to the designated
    * monitor [[akka.actor.typed.ActorRef]] before invoking the wrapped behavior. The
    * wrapped behavior can evolve (i.e. return different behavior) without needing to be
    * wrapped in a `monitor` call again.
+   *
+   * @param id Used to identify when a monitor is added again (using instance equality),
+   *           and the outermost/previous one should be removed. If not called recursively,
+   *           just pass a new `InterceptId` instance.
    */
-  def monitor[T](clazz: Class[T], monitor: ActorRef[T], behavior: Behavior[T]): Behavior[T] = {
-    akka.actor.typed.scaladsl.Behaviors.monitor(monitor, behavior)(reflect.ClassTag(clazz))
+  def monitor[T](clazz: Class[T], monitor: ActorRef[T], behavior: Behavior[T], id: WrappedBehaviorId): Behavior[T] = {
+    akka.actor.typed.scaladsl.Behaviors.monitor(monitor, behavior, id)(reflect.ClassTag(clazz))
```

Comment:
can't we use equals on the `monitor: ActorRef` here instead of requiring the `id`?

Reply:
Yeah, that sounds reasonable, if you have the same recipient you probably want only to monitor to it once.